action.yml

```diff
@@ -244,6 +248,15 @@ runs:
           false:
             kube_version: --kube-version=${{ steps.metadata.outputs.kube_version }}      
 
+    - name: Extra env vars
+      if: ${{ inputs.extra-env-vars != '' }}
+      shell: bash
+      run: |
+        echo '${{ inputs.extra-env-vars }}' > env.json
+        for var in $(jq -r 'keys[]' env.json); do
+          echo "::add-mask::$(jq -r --arg v "$var" '.[$v]' env.json)"
+        done
+        cat env.json | jq -r 'to_entries | .[] | "\(.key)=\(.value)"' >> $GITHUB_ENV
```

Comment:
This doesn't feel right. My gut is saying there would be a lot of headache dealing with the input format as json and then making sure the input works for both those jq commands.

If you're trying to add authentication to private registries for [helmfile registries](https://helmfile.readthedocs.io/en/latest/#oci-registries) could we just use env variables - similar to this but just on the workflow itself?

e.g.
```yaml
steps:
 - uses: cloudposse/github-action-deploy-argocd
   env:
      ARTIFACTORY_USERNAME: foo
      ARTIFACTORY_PASSWORD: ${{ secrets.artifactory-password }}
```
which would then call this entire workflow with the env variables set?
...

Reply:
Yes, there is pain to ensuring the JSON is correct. This is a totem pole.

> A (source repos) -> B (reusable workflow) -> C (this workflow)

A has needs. B does not know of those particular needs that could change wildly per repo so adding a custom env var for every single repos needs can get pretty tedious.

This is indeed not the ideal solution, but we need a way to get from A -> C without tweaking B every time a new app repo changes needs.